core/src/test/java/org/projectnessie/cel/extension/StringsTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2022 The Authors of CEL-Java
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.cel.extension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.cel.Ast;
+import org.projectnessie.cel.Env;
+import org.projectnessie.cel.Library;
+import org.projectnessie.cel.Program;
+import org.projectnessie.cel.common.types.BoolT;
+import org.projectnessie.cel.common.types.pb.ProtoTypeRegistry;
+
+public class StringsTest {
+
+  @Test
+  public void testCharAt() {
+    testExpression("'hello'.charAt(4) == 'o'");
+    testExpression("'hello'.charAt(5) == ''");
+  }
+
+  @Test
+  public void testCharAtNegativeIndex() {
+    assertThatThrownBy(() -> evaluateExpression("'hello'.charAt(-1) == 'h'"))
+        .isExactlyInstanceOf(RuntimeException.class);
+  }
+
+  @Test
+  public void testCharAtIndexOverflow() {
+    assertThatThrownBy(() -> evaluateExpression("'hello'.charAt(65535) == ''"))
+        .isExactlyInstanceOf(RuntimeException.class);
+  }
+
+  @Test
+  public void testIndexOf() {
+    testExpression("'hello mellow'.indexOf('') == 0");
+    testExpression("'hello mellow'.indexOf('ello') == 1");
+    testExpression("'hello mellow'.indexOf('jello') == -1");
+  }
+
+  @Test
+  public void testLastIndexOf() {
+    testExpression("'hello mellow'.lastIndexOf('') == 12");
+    testExpression("'hello mellow'.lastIndexOf('ello') == 7");
+    testExpression("'hello mellow'.lastIndexOf('jello') == -1");
+  }
+
+  @Test
+  public void testLowerAscii() {
+    testExpression("'TacoCat'.lowerAscii() == 'tacocat'");
+    testExpression("'TacoCÆt Xii'.lowerAscii() == 'tacocÆt xii'");
+  }
+
+  @Test
+  public void testReplace() {
+    testExpression("'hello hello'.replace('he', 'we') == 'wello wello'");
+  }
+
+  @Test
+  public void testSplit() {
+    testExpression("'hello hello hello'.split(' ') == ['hello', 'hello', 'hello']");
```

Comment:
Would be nice to have the tests from https://github.com/google/cel-go/pull/316/files#diff-01b0150ddfed956af4270fcad9e73cc6f8d8ea0ad279353349682800208cc5fdR59 here